### PR TITLE
Delete collected data setting UI part 3 (connect #1049)

### PR DESCRIPTION
**Pull request by valllllll2000**
#### Before the PR (what is the issue or what needed to be done)
NA
#### The solution
Part 3 for deleting collected data, using the logic in the UI.
Removed old classes no longer used
Using the previously added dialogs
#### Screenshots (if appropriate)
NA
#### Reviewer Checklist
* [ ] Connect the issue
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

File: app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java [hunk 1]
```diff
@@ -136,5 +149,72 @@ public void destroy() {
...
+    public void deleteCollectedData() {
+        unSyncedTransmissionsExist.execute(new DefaultObserver<Boolean>(){
+            @Override
+            public void onError(Throwable e) {
+                Timber.e(e);
+                view.showDeleteCollectedData();
+            }
+
+            @Override
+            public void onNext(Boolean exist) {
```
Comment (muloem) [inline app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java]: Why not a primitive here? Also maybe use the convention of prefixing with `is...` or `has...`?

File: app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java [hunk 2]
```diff
@@ -136,5 +149,72 @@ public void destroy() {
...
+    public void deleteAllData() {
+        unSyncedTransmissionsExist.execute(new DefaultObserver<Boolean>(){
+            @Override
+            public void onError(Throwable e) {
+                Timber.e(e);
+                view.showDeleteAllData();
+            }
+
+            @Override
+            public void onNext(Boolean exist) {
```
Comment (muloem) [inline app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java]: Same as previous comment of Boolean

File: app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java [hunk 3]
```diff
@@ -136,5 +149,72 @@ public void destroy() {
...
+    }
+
+    private class ClearDataObserver extends DefaultObserver<Boolean> {
+        @Override
+        public void onError(Throwable e) {
+            view.showClearDataError();
+        }
+
+        @Override
+        public void onNext(Boolean cleared) {
```
Comment (muloem) [inline app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java]: Ditto

File: app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java (hunk 1 above)
Comment (valllllll2000) [reply app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java]: 1) We are looking if unsynced transmissions exists or not, I cannot come up with a better name except, isExisting?

2) rxjava uses Boolean better use it. This how it is in all Observers

File: app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java (hunk 2 above)
Comment (valllllll2000) [reply app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java]: rxjava uses Boolean better use it. This how it is in all Observers

File: app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java (hunk 3 above)
Comment (valllllll2000) [reply app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java]: rxjava uses Boolean better use it. This how it is in all Observers

**Pull request merged**